Feat/community page feedback: 커뮤니티 페이지 관련 피드백 반영

nnyouung opened the pull request:
### 개요
커뮤니티 페이지 관련 피드백 반영
<br>

### 수정사항
- 댓글 삭제 api 연결 및 댓글 조회 api 형식 변경에 따른 수정
- 이미지 확대 페이지 및 관련 기능 구현
- 커뮤니티 페이지의 이미지 업로드 오류 해결

~~커뮤니티 페이지에 이미지를 올리면 expo 앱 자체가 다운되는 문제가 있었는데요. 알고 보니까 지금 사용하는 expo managed workflow에서는 다중 이미지 선택 자체를 지원하지 않는다고 합니다. Bare Workflow로 전환한다면 사용할 수 있는데, 그걸 지금 바꾸기에는 더 고민해봐야 할 거 같아서요ㅜㅜ 공식 라이브러리 뿐만 아니라 사람들이 많이 사용하는 서드파티 라이브러리들도 현재 환경에서는 사용할 수 있는 게 없는 거 같습니다.. 서드파티에 대해서는 더 알아보긴 할 예정인데, 일단 임시방편으로 한 장씩 가져올 수 있게 구현해놨습니다.~~
-> 'fix: WritePage.jsx를 이전 상태로 복구' 커밋에서 해결하였습니다.

자세한 내용은 커밋 참고 부탁드립니다!
<br>

### 테스트 화면
댓글 삭제 api 연결 및 댓글 조회 api 형식 변경에 따른 수정

https://github.com/user-attachments/assets/e3ee9456-226e-46b9-8ec6-20da544fcf87

<br><br>

이미지 확대 페이지 및 관련 기능 구현
-> 저장 시 사진첩에 들어가는 것까지 확인했습니다!

https://github.com/user-attachments/assets/79d0d386-a63f-4eb6-a305-a10737ef03c4

<br><br>

커뮤니티 페이지의 이미지 업로드 오류 해결 

https://github.com/user-attachments/assets/7d39f681-2e6b-4ae0-9a12-380926ef16ab

https://github.com/user-attachments/assets/766788a4-a9c2-4e5f-8dcb-4c79fbadd489


<br><br>

nnyouung committed:
feat: 댓글 삭제 api 연결 및 댓글 조회 api 형식 변경에 따른 수정

수정사항
- api.js
  - 댓글 삭제 api 연결 추가
  - 댓글 조회 api 형식 변경에 따른 엔드포인트 수정
- ModalKebabMenu.jsx, ItemComment.jsx
  - 댓글 삭제 백엔드 연결
- PostPage.jsx
  - 댓글 삭제 후 UI가 실시간으로 업데이트될 수 있게, comments 변경 시 자동 반영되도록 수정

nnyouung committed:
feat: 이미지 확대 페이지 및 관련 기능 구현

수정사항
- App.jsx
  - 'EnlargeImagePage'를 앱 내에 등록
- PostPage.jsx, PostStyles.jsx
  - 이미지가 여러 장일 때의 세부 스타일 수정
  - 이미지 클릭 시 'EnlargeImagePage'로 이동할 수 있게끔 연결
- IconImageDownload.jsx, Download.svg
  - '이미지 저장하기' 버튼 아이콘 컴포넌트 제작
- EnlargeImagePage.jsx, EnlargeImageStyles.jsx
  - 이미지 확대 페이지 구현
  - 이미지 저장하기 기능 구현

nnyouung committed:
fix: 커뮤니티 페이지의 이미지 업로드 오류 해결

수정사항
- api.js
  - 이미지가 리스트로 전달되므로, 이에 맞춰 폼데이터 형식 수정
- WritePage.jsx
  - expo managed workflow의 다중 이미지 선택 불가 문제로, 이미지를 한 장씩 선택하여 리스트에 추가하는 형식으로 수정

seungholee-dev commented:
Base branch를 dev 브랜치로 바꿔놨습니다 :)

seungholee-dev commented on src/components/community/ItemComment.jsx:
```diff
@@ -117,15 +117,23 @@ const ItemComment = ({ commentList = [], onReply }) => {
 
 	const [myMemberId, setMyMemberId] = useState(null);
 
+	const getMyId = async () => {
+		try {
+			const id = await getMyMemberId();
+			setMyMemberId(id);
+		} catch (error) {
+			console.error(
+				"내 멤버 아이디 가져오기 실패:",
+				error.response ? error.response.data : error.message,
+			);
+		}
+	};
+
```
이 부분은 기존의 getMyMemberID 함수를 랩핑하지 않고 그대로 import해서 사용하면 될 것 같은데 특별한 이유가 있을까요?!

nnyouung replied:
랩핑이라는 말이 왜 다시 useState로 선언했냐는 말씀 맞으실까요?!
memberId는 renderComment에서 필요한데요, 이 함수가 렌더링 과정에서 호출됩니다. 렌더링 과정은 동기적이기 때문에 비동기 함수를 직접 사용할 수 없어 useEffect를 이용하여 처리하였습니다..!

seungholee-dev replied:
제가 조금 애매하게 말했군요!
getMyId 함수 내부에서, `try catch`가 불필요한 것 같아서요! 제 기억으로 getMyMemberId 자체에서 try catch가 있는 것으로 알고 있어서요!

nnyouung replied:
앗 그 말씀이셨군요. 불필요한 게 맞네요!! 수정해서 커밋 올려놨습니다!

seungholee-dev reviewed (approved):
공유 감사합니다!

그런 문제가 있었군요..!
expo에 의존하고 있는 것들이 많아서 현재 당장 바꾸기는 어려울 것 같은데요, 혹시 
아래의 옵션을 사용해도 안되는 부분일까요?! 

https://docs.expo.dev/versions/latest/sdk/imagepicker/#:~:text=false-,allowsMultipleSelection,-(optional)

nnyouung commented:
> 공유 감사합니다!
> 
> 그런 문제가 있었군요..! expo에 의존하고 있는 것들이 많아서 현재 당장 바꾸기는 어려울 것 같은데요, 혹시 아래의 옵션을 사용해도 안되는 부분일까요?!
> 
> https://docs.expo.dev/versions/latest/sdk/imagepicker/#:~:text=false-,allowsMultipleSelection,-(optional)

저 옵션이 원래 제가 사용하던 옵션인데요............
공식 문서에 있는데, 안 되는 게 이상하여 혹시나 하는 마음에 다시 전 코드로 돌아가니 작동합니다...........
중간에 업데이트한 것도 없는데 이유를 모르겠네요.................

https://expo.canny.io/feature-requests/p/allow-choosing-multiple-images-in-imagepicker
이게 오류 해결할 당시에 본 글 중 하나인데, 좀 오래된 글이지만 안 될 이유가 없어서 해당 문제인 줄만 알았습니다ㅜㅜ
그래도 해결되어 다행이네요!! 다시 전 코드로 돌려놓고 커밋 올리겠습니다 ㅎㅎ 감사합니다!

nnyouung committed:
fix: WritePage.jsx를 이전 상태로 복구

수정사항
- 'fix: 커뮤니티 페이지의 이미지 업로드 오류 해결'에서 수정한 부분을 원상복구
- 커뮤니티 페이지에서 이미지 비율 제한(aspect)이 필요하지 않아 해당 부분 삭제

seungholee-dev commented:
> 이게 오류 해결할 당시에 본 글 중 하나인데, 좀 오래된 글이지만 안 될 이유가 없어서 해당 문제인 줄만 알았습니다ㅜㅜ
그래도 해결되어 다행이네요!! 다시 전 코드로 돌려놓고 커밋 올리겠습니다 ㅎㅎ 감사합니다!

그렇군요! 잘 동작한다니 다행이네요! :raised_hands:

nnyouung committed:
refactor: 불필요한 try/catch문 삭제
